### PR TITLE
Add Advanced Dropdown + 2 Options

### LINKED ISSUE #12: Expose more dnsPacket features in web interface
* DNSSEC_OK and ECS seem to be supported by dohjs already so those should be fairly easy.
* Bonus features would be adding arbitrary EDNS option (similar to dig with (opt_code, payload) tuples)
* Anything else outside of EDNS that could be beneficial? It gets tricky with adding features, but not going overboard allowing manipulation of every bit
* Could probably put these in an "advanced" drop down

### LINKED ISSUE #37: Optionally save interface parameters to local storage
This would involve saving fields, such as the resolver URL and HTTPS method to local storage. This should be enabled via a checkbox (possible under the advanced dropdown in #12).

### PR DESCRIPTION
**Pull request by jacobgb24**
* Adds an "Advanced" dropdown with 2 switches
  * First switch saves all form values to local storage (except qname)
  * second switch enables DNSSEC
* Made cloudflare the default resolver and removed validation check on url
* other minor UI changes, including moving qname to top

Closes #12 (not sure why this was closed early)
Closes #37

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -57,26 +57,35 @@ document.addEventListener('DOMContentLoaded', function(e) {
...
-          .then(successFunction)
-          .catch(errorFunction);
+
+        const query = doh.makeQuery(qname, qtype);
+        if (dnssecSwitch.checked) {
+            query.additionals =  [{
+                type: 'OPT',
+                name: '.',
+                udpPayloadSize: 4096,
+                flags: 1 << 15, // DO bit
```
Comment (jacobgb24) [inline main.js:78]: @kimbo I tried using `doh.DNSSEC_OK` here (which my IDE even suggested) but it was undefined when testing.

File: main.js [hunk 2]
```diff
@@ -57,26 +57,35 @@ document.addEventListener('DOMContentLoaded', function(e) {
...
+                options: []
+            }]
+        }
+        console.log(JSON.stringify(query));
+        console.log();
+        console.log(query.flags & doh.AUTHENTIC_DATA);
+
+        doh.sendDohMsg(query, url, method)
```
Comment (jacobgb24) [inline main.js:86]: Not matching the actual signature here which takes 5 params (headers + timeout missing). I can set the headers to null, but not timeout. What's appropriate there?

File: localstorage.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+const saveSwitch = document.getElementById('save-switch');
+
+document.addEventListener('DOMContentLoaded', function(e) {
+    saveSwitch.addEventListener("change", handleSaveSwitch);
+    // try loading everything. If anything loads, set the save switch
+    formsOfInterest.forEach((id) => {
+        const elem = document.getElementById(id);
+        const val = localStorage.getItem(id);
+        if (val !== null ) {
+            elem.type === "checkbox" ? elem.checked = val : elem.value = val;
```
Comment (kimbo) [inline localstorage.js:18]: We should change this so `val` is properly converted to boolean.
Alternatively, we could treat them like environment variables (i.e. [here](https://github.com/byu-imaal/dohjs/pull/45/files#diff-6dc4294a01cc7cb81592c8be9d96130dR62) we save `false` as `''` and `true` as anything that's not empty string)

File: main.js (hunk 1 above)
Comment (kimbo) [reply main.js:78]: We'll have to expose `dnsPacket` in dohjs, which will make things a lot more flexible, actually.

File: main.js (hunk 2 above)
Comment (kimbo) [reply main.js:86]: I think it's fine as is. We can add those in later if we feel like it. (JavaScript doesn't care)

**Pull request merged**